### PR TITLE
Fix default format(float) result when the number is too big

### PR DESCRIPTION
**Pull request by ChJR**
Before this change, format(1e16) returns '10000000000000000.0' in RustPython while CPython 3.7 returns '1e+16'.
After this change, format(1e16) returns '1e+16' in RustPython too.

### CODE REVIEW

File: vm/src/format.rs [hunk 1]
```diff
@@ -405,7 +406,7 @@ impl FormatSpec {
                     magnitude if magnitude.is_infinite() => Ok("inf".to_owned()),
                     // Using the Debug format here to prevent the automatic conversion of floats
                     // ending in .0 to their integer representation (e.g., 1.0 -> 1)
```
Comment (coolreader18) [inline vm/src/format.rs:408]: You could probably remove this explanation, since it doesn't apply anymore.

**Review by coolreader18 — APPROVED** [review]
Nice work!

File: vm/src/format.rs (hunk 1 above)
Comment (ChJR) [reply vm/src/format.rs:408]: Good points!

**Comment by coolreader18** [comment]
Thanks for contributing!

**Pull request merged**